lib/init.ts

```diff
@@ -84,3 +91,28 @@ You can visit https://algolia.com/events/debugger instead.`);
     this.setAnonymousUserToken();
   }
 }
+
+type ThisParams = {
+  _userHasOptedOut: InitParams["userHasOptedOut"];
+  _useCookie: InitParams["useCookie"];
+  _cookieDuration: InitParams["cookieDuration"];
+  _region: InitParams["region"];
+};
+
+function setOptions(
+  target: ThisParams,
+  { patch, ...options }: InitParams,
+  defaultValues: ThisParams
+) {
+  if (!patch) {
+    Object.assign(target, defaultValues);
+  }
+
+  for (const key in options) {
```

Comment:
for in needs a `hasOwnProperty` check. Can we use Object.assign or is that too low browser support?

Reply:
`Object.assign` is fine, we have a polyfill loaded.

Is https://github.com/algolia/search-insights.js/pull/401/commits/2a3cd27a282fd426085b642b10706350965dccbd what you had in mind?